prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+            organizationName = organizationName,
+            format = Sender.Format.CSV,
+            customerStatus = CustomerStatus.INACTIVE,
+            schemaName = "one"
+        )
+        settings.senderStore[sender.fullName] = sender
+        val sender2 = CovidSender(
+            name = "default",
+            organizationName = otherOrganizationName,
+            format = Sender.Format.CSV,
+            customerStatus = CustomerStatus.INACTIVE,
+            schemaName = "one"
+        )
+        settings.senderStore[sender2.fullName] = sender2
+
+        val receiver = Receiver(
+            "elr-secondary",
+            organizationName,
+            Topic.FULL_ELR,
+            CustomerStatus.TESTING,
+            "schema1"
+        )
+        settings.receiverStore[receiver.fullName] = receiver
+
+        val receiver2 = Receiver(
+            "test-lab-2",
+            otherOrganizationName,
+            Topic.FULL_ELR,
+            CustomerStatus.TESTING,
+            "schema1"
+        )
+        settings.receiverStore[receiver2.fullName] = receiver2
+
+        mockkObject(OktaAuthentication.Companion)
+        every { OktaAuthentication.Companion.decodeJwt(any()) } returns
+            TestDefaultJwt(
+                "a.b.c",
+                Instant.now(),
+                Instant.now().plusSeconds(60),
+                claimsMap
+            )
+
+        return SettingsFunction(
+            settingsFacade = facade
+        )
+    }
+
+    private fun setupHttpRequestMessageForTesting(method: HttpMethod = HttpMethod.GET): MockHttpRequestMessage {
+        val httpRequestMessage = MockHttpRequestMessage("{}", method)
+        httpRequestMessage.httpHeaders += mapOf(
+            "authorization" to "Bearer 111.222.333",
+            "content-type" to HttpUtilities.jsonMediaType
+        )
+
+        return httpRequestMessage
+    }
+
+    @Test
+    fun `test access user cannot view full organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view all organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot view their organization's settings (getOneOrganization)`() {
```

Comment:
I think this test is a little off, a user _is_ able t to access there organization.  The reason this is passing is that you are passing a receiver in on 214L rather than the organization name

Reply:
Thanks to this I was able to understand what the issue was with the claims. It passes correctly now!